common/src/main/java/org/opensearch/sdk/SdkClientUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.action.ActionFuture;
+import org.opensearch.common.util.concurrent.FutureUtils;
+import org.opensearch.common.util.concurrent.UncategorizedExecutionException;
+
+public class SdkClientUtils {
+
+    /**
+     * Unwraps the cause of a {@link CompletionException}. If the cause is a subclass of {@link RuntimeException}, rethrows the exception.
+     * Otherwise wraps it in an {@link OpenSearchException}. Properly re-interrupts the thread on {@link InterruptedException}.
+     * @param e the CompletionException
+     * @return The wrapped cause of the completion exception if unchecked, otherwise an OpenSearchException wrapping it.
+     */
+    public static RuntimeException unwrapAndConvertToRuntime(CompletionException e) {
+        Throwable cause = e.getCause();
+        if (cause instanceof InterruptedException) {
+            Thread.currentThread().interrupt();
+        }
+        // Below is the same as o.o.ExceptionsHelper.convertToRuntime but cause is a throwable
+        if (cause instanceof RuntimeException) {
+            return (RuntimeException) cause;
+        }
+        return new OpenSearchException(cause);
+    }
+
+    /**
+     * Get the original exception of an {@link UncategorizedExecutionException} with two levels of cause nesting.
+     * Intended to recreate the root cause of an exception thrown by {@link ActionFuture#actionGet}, which was handled by {@link FutureUtils#rethrowExecutionException}.
+     * @param throwable a throwable with possibly nested causes
+     * @return the root cause of an ExecutionException if it was not a RuntimeException, otherwise the original exception 
+     */
+    public static Throwable getRethrownExecutionExceptionRootCause(Throwable throwable) {
+        if (throwable instanceof UncategorizedExecutionException && throwable.getCause() instanceof ExecutionException) {
+            return throwable.getCause().getCause();
+        }
+        return throwable;
```

Comment:
Shouldn't this be `throwable.getCause()`?

Reply:
Nope, `throwable` is an `UncategorizedExecutionException` and `throwable.getCause()` is an `ExecutionException`.  So the original exception is in two levels of nesting.

Probably not the original intent but that's how the exception wrapping in `actionGet()` works.

The test class makes it clear/simple with one RuntimeException and one not (IOException).

Reply:
The double-wrapped causes is what you implemented [here](https://github.com/opensearch-project/ml-commons/blob/feature/multi_tenancy/plugin/src/main/java/org/opensearch/ml/action/tasks/DeleteTaskTransportAction.java#L156-L162).  The difference is that I'm very specifically targeting the 2-layer nesting from actionGet vs. a general arbitrarily deep nesting.

Reply:
Also coverage report shows both branches are covered by the two tests:
![image](https://github.com/opensearch-project/ml-commons/assets/9291703/de2d8d38-d8d9-44d6-9a94-a67888100c75)